framework/src/processing/README.md

```diff
@@ -48,17 +48,54 @@ The constructs creates an EKS cluster, install the necessary controllers and ena
  * An EKS cluster (VPC configuration can be customized)
  * A tooling nodegroup to run tools to run controllers
  * Kubernetes controlers: EBS CSI Driver, Karpenter, ALB Ingress Controller, cert-manager  
- * Optionally Default Kaprenter NodePools and EC2NodeClass as listed [here](https://github.com/awslabs/data-solutions-framework-on-aws/tree/main/framework/src/processing/lib/spark-runtime/emr-containers/resources/k8s/karpenter-provisioner-config).
+ * Optionally Default Kaprenter `NodePools` and `EC2NodeClass` as listed [here](https://github.com/awslabs/data-solutions-framework-on-aws/tree/main/framework/src/processing/lib/spark-runtime/emr-containers/resources/k8s/karpenter-provisioner-config).
+ * An Amazon S3 bucket to store the pod templates for the `NodePools` created above.
 
-The construct will upload on S3 the Pod templates required to run EMR jobs on the default Kaprenter NodePools and EC2NodeClass. It will also parse and store the configuration of EMR on EKS jobs for each default nodegroup in object parameters.
+The construct can be configured by customizing the [properties](https://constructs.dev/packages/aws-dsf/v/1.0.0-rc.5/api/SparkEmrContainersRuntimeProps?lang=python&submodule=processing) that are passed to the constructor.
 
+### EC2 Capacity
+
+The EC2 capacity to execute the jobs is defined with [Karpenter](https://karpenter.sh/docs/getting-started/) `NodePools` and `EC2NodeClass`, these are always created by the construct, you can opt out from their creation by setting the [`default_nodes`](https://constructs.dev/packages/aws-dsf/v/1.0.0-rc.5/api/SparkEmrContainersRuntimeProps?lang=python&submodule=processing#property.defaultNodes) to `False`. The construct uploads to an S3 buckets (created by the consturct) the Pod templates required to run EMR jobs on the default Kaprenter `NodePools` and `EC2NodeClass`. It will also parse and store the configuration of EMR on EKS jobs for each default nodegroup in object parameters. The construct provides the S3 locations for the pod templates as a cloudformation.
+
+### Execution role
+
+The execution role is the IAM role that is used by job to access AWS resources, this can be for example an S3 bucket that is storing data or to which the job write the data. The construct has a method which allows the creation of an IAM role that is onboarded to be used in the EKS cluster and from the namespace where the virtual cluster is created. The method attach an IAM polic provided by the user and also add a policy to access the pod templates to use the default EC2 capacity defined above.
 
 ## Usage
 
+The constructs offers multiple methods to streamline the creation of runtimes, these methods are:
```

Comment:
move the methods outside of `Usage`. 
`CreateExecutionRole` should be in the `execution role` part.
`AddkarpenterNodePoolAndNodeClass` should be in the `EC2 capacity`part.
`AddEmrVirtualCluster` should be in `Isolating workloads on a shared cluster` part.

For each of them, we should provide a code snippet.